source/opt/scalar_replacement_pass.h

```diff
@@ -39,7 +39,9 @@ class ScalarReplacementPass : public MemPass {
       : max_num_elements_(limit) {
     name_[0] = '\0';
     strcat(name_, "scalar-replacement=");
-    sprintf(&name_[strlen(name_)], "%d", max_num_elements_);
+    const size_t name_len = strlen(name_);
+    snprintf(&name_[name_len], sizeof(name_) - name_len, "%d",
```

Comment:
See #4815 for a better fix.

Note:  This   sizeof(name_) - name_len does not leave room for the additional NUL character that would be appended if there were enough room.

Basically,  if your buffer is 10 bytes, and you try to put 10 bytes of content in there, then only 9 bytes will be put in, because it has to put in the terminating  NUL character.

Please note this @strangewiz because you may be making this mistake in other patches to other projects.